### DISCUSSION TITLE
Add line and column number to AlphaTab errors

### DISCUSSION THREAD

**Discussion by jonaro00** [CoderLine/alphaTab#1044] [Ideas]
In addition to the current offset from the start of the input string, errors can point out more precisely where a parsing error happened by having AlphaTexImporter keep count of how many newlines has passed. This can be used in live editors (similar to playground) to help the user see where the syntax error is.

The error message logic can also be expanded upon, and more detailed error messages can be added.

I plan on implementing this myself. Feedback, discussion and more ideas are appreciated 😄

**Comment by Danielku15** [comment]
There is no main reason why we do not have line numbers until now, maybe just laziness on implementing it 😆 But counting the numbers of `\n` in the `AlphaTexImporter.newSy` should be fairly simple to add.
